internal/services/monitor/monitor_workbook_resource.go

```diff
@@ -0,0 +1,249 @@
+				}
+			}
+
+			return nil
+		}),
+	}
+}
+
+func resourceMonitorWorkbookCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Monitor.WorkbookClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := workbook.NewWorkbookID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+
+	if d.IsNewResource() {
+		existing, err := client.WorkbooksGet(ctx, id, workbook.WorkbooksGetOperationOptions{CanFetchContent: utils.Bool(true)})
+		if err != nil {
+			if !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+		}
+
+		if !response.WasNotFound(existing.HttpResponse) {
+			return tf.ImportAsExistsError("azurerm_monitor_workbook", id.ID())
+		}
+	}
+
+	kindValue := workbook.WorkbookSharedTypeKindShared
+	props := workbook.Workbook{
+		Kind:     &kindValue,
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Properties: &workbook.WorkbookProperties{
+			Category:       d.Get("category").(string),
+			Description:    utils.String(d.Get("description").(string)),
+			DisplayName:    d.Get("display_name").(string),
+			SerializedData: d.Get("serialized_data").(string),
+		},
+
+		Tags: tagsHelper.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if storageURI, ok := d.GetOk("storage_uri"); ok {
+		props.Properties.StorageUri = utils.String(storageURI.(string))
+	}
+
+	identity, err := identity.ExpandLegacySystemAndUserAssignedMap(d.Get("identity").([]interface{}))
+	if err != nil {
+		return fmt.Errorf("expanding `identity`: %+v", err)
+	}
+
+	props.Identity = identity
+
+	if _, err := client.WorkbooksCreateOrUpdate(ctx, id, props, workbook.WorkbooksCreateOrUpdateOperationOptions{SourceId: utils.String(d.Get("source_id").(string))}); err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+	return resourceMonitorWorkbookRead(d, meta)
+}
+
+func resourceMonitorWorkbookRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Monitor.WorkbookClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := workbook.ParseWorkbookID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.WorkbooksGet(ctx, *id, workbook.WorkbooksGetOperationOptions{CanFetchContent: utils.Bool(true)})
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	d.Set("name", id.ResourceName)
+	d.Set("resource_group_name", id.ResourceGroupName)
+
+	if model := resp.Model; model != nil {
+		d.Set("location", location.Normalize(*model.Location))
```

Comment:
there's a crash here, this can become:

```suggestion
		d.Set("location", location.NormalizeNilable(model.Location))
```